src/__tests__/type-modifiers.js

```diff
@@ -353,6 +353,37 @@ test('{shift}a{/shift}', () => {
   `)
 })
 
+test('{capslock}a{/capslock}', () => {
+  const {element, getEventSnapshot} = setup('<input />')
+
+  userEvent.type(element, '{capslock}a{/capslock}')
+
+  expect(getEventSnapshot()).toMatchInlineSnapshot(`
+    Events fired on: input[value="a"]
+
+    input[value=""] - pointerover
+    input[value=""] - pointerenter
+    input[value=""] - mouseover: Left (0)
+    input[value=""] - mouseenter: Left (0)
+    input[value=""] - pointermove
+    input[value=""] - mousemove: Left (0)
+    input[value=""] - pointerdown
+    input[value=""] - mousedown: Left (0)
+    input[value=""] - focus
+    input[value=""] - focusin
+    input[value=""] - pointerup
+    input[value=""] - mouseup: Left (0)
+    input[value=""] - click: Left (0)
+    input[value=""] - keydown: CapsLock (20)
```

Comment:
Ah, this one is tricky. Basically for caps lock, it should keydown *and* keyup here, but preserve the modifier state. Hmmm...

Reply:
So, do you think a PR will also be needed to fireEvent for the case of CapsLock?

Reply:
No, `fireEvent` is all about dispatching individual events in complete isolation. It is not concerned about state. This will only need to happen here in user-event. It's possible we won't be able to re-use the existing `createModifierCallbackEntries` for this one.

Reply:
Scratch that. Updated with a switch/case. I'm not sure if it's initially beneficial to write it that way, but I didn't know if other modifier keys might also require the behavior at some point.

Reply:
Saw the comment after. I can write a separate function for it if preferred?

Reply:
@Sonic12040, could you try removing the custom logic and inlining it for the caps lock. I think that will result in simpler code in this case. I'm worried that this is a hasty abstraction ([AHA Programming](https://kcd.im/aha)) and I think the small amount of duplication will be better.

Reply:
Updated with a duplicate function that performs the CapsLock behavior.